--- origsrc/luafilesystem-1.5.0/Makefile	2009-10-20 15:54:35.000000000 -0500
+++ src/luafilesystem-1.5.0/Makefile	2011-02-12 22:48:05.346018500 -0600
@@ -12,7 +12,7 @@ OBJS= src/$T.o
 lib: src/lfs.so
 
 src/lfs.so: $(OBJS)
-	MACOSX_DEPLOYMENT_TARGET="10.3"; export MACOSX_DEPLOYMENT_TARGET; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS)
+	MACOSX_DEPLOYMENT_TARGET="10.3"; export MACOSX_DEPLOYMENT_TARGET; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS) -llua
 
 install:
 	mkdir -p $(LUA_LIBDIR)
--- origsrc/luafilesystem-1.5.0/src/lfs.c	2009-10-20 15:54:35.000000000 -0500
+++ src/luafilesystem-1.5.0/src/lfs.c	2011-02-12 22:49:41.488986100 -0600
@@ -98,8 +98,12 @@ typedef struct dir_data {
  #endif
 #define STAT_FUNC _stati64
 #else
+#ifndef _O_TEXT
 #define _O_TEXT               0
+#endif
+#ifndef _O_BINARY
 #define _O_BINARY             0
+#endif
 #define lfs_setmode(L,file,m)   ((void)((void)file,m),  \
 		 luaL_error(L, LUA_QL("setmode") " not supported on this platform"), -1)
 #define STAT_STRUCT struct stat
